tools/jenkinshelper-main.ksh

```diff
@@ -60,7 +60,7 @@ stage_build_changed() {
 	for f in $(git diff --name-only HEAD^1 | grep Makefile; exit 0); do
 		echo "jenkinshelper: building for ${f%/*}..."
 		curpwd=$(pwd)
-		cd "${f%/*}" && COMPONENT_BUILD_ARGS=-j$(psrinfo -t -c) gmake publish
+		cd "${f%/*}" && gmake clean && COMPONENT_BUILD_ARGS=-j$(psrinfo -t -c) gmake publish
```

Comment:
@AndWac, is this file really used on build server(s)?  The `COMPONENT_BUILD_ARGS` setting here is plain wrong.

Reply:
No, it is not used there. I guess @Toasterson  is using it on the PR checker.

Reply:
No, this file was intended to replace the existing main jenkins builder but we never got it to a state where people where happy.

The PR builder uses https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/ci/pr-builder.groovy and thus https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/tools/jenkinshelper.ksh

gmake clean is not executed during build, as every PR builder should have an own workspace.

Reply:
To be more explicit about what I'm trying to fix:  when a patch in a pull request is updated it appears that the tree is reused and you get errors like:
```
+ cd components/python/python-312
+ gmake publish
+ psrinfo -t -c
+ COMPONENT_BUILD_ARGS=-j32
/usr/gnu/bin/patch -d /ws/jenkins/workspace/rganisation_oi-userland_PR-15955/components/python/python-312/Python-3.12.1 --strip=1 --backup --version-control=numbered < patches/00-include-ncurses.patch
patching file configure.ac
patching file configure.ac
/usr/bin/touch /ws/jenkins/workspace/rganisation_oi-userland_PR-15955/components/python/python-312/Python-3.12.1/.patched-00-include-ncurses.patch
/usr/gnu/bin/patch -d /ws/jenkins/workspace/rganisation_oi-userland_PR-15955/components/python/python-312/Python-3.12.1 --strip=1 --backup --version-control=numbered < patches/00-socket-test-2.patch
patching file Modules/Setup.stdlib.in
Reversed (or previously applied) patch detected!  Assume -R? [n] 
Apply anyway? [n] 
Skipping patch.
2 out of 2 hunks ignored -- saving rejects to file Modules/Setup.stdlib.in.rej
gmake: *** [/ws/jenkins/workspace/rganisation_oi-userland_PR-15955/make-rules/prep-patch.mk:104: /ws/jenkins/workspace/rganisation_oi-userland_PR-15955/components/python/python-312/Python-3.12.1/.patched-00-socket-test-2.patch] Error 1
+ rc=2
+ cd /ws/jenkins/workspace/rganisation_oi-userland_PR-15955
+ echo 'jenkinshelper: done with components/python/python-312 return code 2'
jenkinshelper: done with components/python/python-312 return code 2

```
( from 
https://jenkins.openindiana.aurora-opencloud.org/blue/organizations/jenkins/OpenIndiana%20Organisation%2Foi-userland/detail/PR-15955/5/pipeline which is from a revision to my python 3.12 pull request)

Reply:
Ah, yes that is correct. Unfortunately, you edited the wrong copy of the file. This is not the PR helper script but the main branch builder script.